watertap/unit_models/electrodialysis_1D.py

```diff
@@ -778,6 +1070,497 @@ def eq_recovery_mass_H2O(self, t):
+            # This is mathematically valid as the ratio eliminate the difference and electro-neutrality stands.
+
+    def _make_performance_dl_polarization(self):
+
+        self.potential_nonohm_dl_x = Var(
+            self.membrane_set,
+            self.flowsheet().time,
+            self.diluate.length_domain,
+            initialize=0.01,  # to inspect
+            bounds=(-50, 50),
+            units=pyunits.volt,
+            doc="Nonohmic potential in two diffusion layers on the two sides of a membrane",
+        )
+
+        self.potential_ohm_dl_x = Var(
+            self.membrane_set,
+            self.flowsheet().time,
+            self.diluate.length_domain,
+            initialize=0.01,  # to inspect
+            bounds=(0, 50),
+            units=pyunits.volt,
+            doc="Ohmic potential in two diffusion layers on the two sides of a membrane",
+        )
+
+        self.dl_thickness_x = Var(
+            self.membrane_set,
+            self.electrode_side_set,
+            self.flowsheet().time,
+            self.diluate.length_domain,
+            initialize=0.0005,
+            bounds=(0, 1e-2),
+            units=pyunits.m,
+            doc="Thickness of the diffusion layer",
+        )
+
+        @self.Constraint(
+            self.flowsheet().time,
+            self.diluate.length_domain,
+            doc="Calculate length-indexed limitting current density",
+        )
+        def eq_current_dens_lim_x(self, t, x):
+            if (
+                self.config.limiting_current_density_method
+                == LimitingCurrentDensityMethod.InitialValue
+            ):
+                return self.current_dens_lim_x[t, x] == (
+                    self.config.limiting_current_density_data
+                    * pyunits.amp
+                    * pyunits.meter**-2
+                    / sum(
+                        self.diluate.properties[t, 0].conc_mol_phase_comp["Liq", j]
+                        for j in self.cation_set
+                    )
+                    * sum(
+                        self.diluate.properties[t, x].conc_mol_phase_comp["Liq", j]
+                        for j in self.cation_set
+                    )
+                )
+
+        @self.Constraint(
+            self.membrane_set,
+            self.electrode_side_set,
+            self.flowsheet().time,
+            self.diluate.length_domain,
+            self.config.property_package.ion_set,
+            doc="Establish relationship between interfacial concentration polarization ratio and current density",
+        )
+        def eq_conc_polarization_ratio(self, mem, side, t, x, j):
+            if (mem == "cem" and side == "cathode_left") or (
+                mem == "aem" and side == "anode_right"
+            ):
+                return self.conc_mem_surf_mol_x[
+                    mem, side, t, x, j
+                ] / self.concentrate.properties[t, x].conc_mol_phase_comp["Liq", j] == (
```

Comment:
Might it be safer to multiply the denominator to the other side rather than keep here in the denominator? The same things goes for your alternative constraint below where you divide by diluate on the LHS.